src/error_codes.rs

```diff
@@ -0,0 +1,64 @@
+use thiserror::Error;
+
+/// Revert with this value for a failing call to `std::revert::require`.
+pub const FAILED_REQUIRE_SIGNAL: u64 = 0xffff_ffff_ffff_0000;
+
+/// Revert with this value for a failing call to `std::token::transfer_to_address`.
+pub const FAILED_TRANSFER_TO_ADDRESS_SIGNAL: u64 = 0xffff_ffff_ffff_0001;
+
+/// Revert with this value for a failing call to `std::message::send_message`.
+pub const FAILED_SEND_MESSAGE_SIGNAL: u64 = 0xffff_ffff_ffff_0002;
+
+/// Revert with this value for a failing call to `std::assert::assert_eq`.
+pub const FAILED_ASSERT_EQ_SIGNAL: u64 = 0xffff_ffff_ffff_0003;
+
+/// Revert with this value for a failing call to `std::assert::assert`.
+pub const FAILED_ASSERT_SIGNAL: u64 = 0xffff_ffff_ffff_0004;
+
+#[derive(Error, Debug)]
+pub enum ErrorSignal {
+    #[error("Failing call to `std::revert::require`")]
+    Require,
+    #[error("Failing call to `std::token::transfer_to_address`")]
+    TransferToAddress,
+    #[error("Failing call to `std::message::send_message`")]
+    SendMessage,
+    #[error("Failing call to `std::assert::assert_eq`")]
+    AssertEq,
+    #[error("Failing call to `std::assert::assert`")]
+    Assert,
+    #[error("Unknown error signal")]
+    Unknown,
```

Comment:
I would maybe remove this `Unknown` variant in favour of changing `from_revert_code` to something like:

```rust
pub fn try_from_revert_code(revert_code: u64) -> Result<Self, UnknownSignal>
```